syntax = 'proto3';

package Proto;

enum PlayerType {
    OWNER = 0;
    GUEST = 1;
};

enum PieceColor {
    WHITE = 0;
    BLACK = 1;
};

message Empty {
};

message Bool {
    bool value = 1;
};

message Integer {
    int32 value = 1;
};

message String {
    string value = 1;
};

message Position {
    int32 x = 1;
    int32 y = 2;
};

message RoomsInfo {
    repeated string names = 1;
}

message RoomSettings {
    string name = 1;
    string password = 2;
};

message RoomWithUsername {
    RoomSettings room = 1;
    string username = 2;
};

message ReadyRequest {
    RoomSettings room = 1;
    PlayerType playertype = 2;
    bool isready = 3;
};

message MoveRequest {
    RoomSettings room = 1;
    Position oldPosition = 2;
    Position newPosition = 3;
};

message LastMoveInfo {
    Position oldPosition = 1;
    Position newPosition = 2;
};
